contracts/VestingUpgradeable_V2.sol

```diff
@@ -0,0 +1,194 @@
+    /// @param _investors the array of investors
+    /// @param _amounts the amount of tokens
+    /// @param _allocationType the type allocation of vesting
+    function addInvestors(
+        address[] calldata _investors,
+        uint256[] calldata _amounts,
+        AllocationType _allocationType
+    ) external virtual override onlyOwner {
+        require(block.timestamp < initialTimestamp, "The vesting was start");
+        uint256 length = _investors.length;
+        require(
+            length != 0 && length == _amounts.length,
+            "Arrays different length"
+        );
+        uint256 totalAmount = 0;
+        for (uint256 i; i < length; i++) {
+            userTokens[_investors[i]][_allocationType] = _amounts[i];
+            totalAmount += _amounts[i];
+            investors[lastIndexInvestors] = _investors[i];
+            lastIndexInvestors++;
+        }
+        rewardToken.mint(address(this), totalAmount);
+        emit AddInvestors(_investors, _amounts, _allocationType);
+    }
+
+    /// @notice Transfer accrued reward tokens to investor
+    /// @dev Transfer accrued reward tokens to investor
+    function withdrawTokens() external virtual override {
+        require(
+            initialTimestamp != 0 && block.timestamp > initialTimestamp,
+            "The vesting was not start"
+        );
+        uint256 amountToTransfer;
+        if (userTokens[msg.sender][AllocationType.Seed] > 0) {
+            amountToTransfer += _calculateUnlock(
+                msg.sender,
+                AllocationType.Seed
+            );
+            rewardsPaid[msg.sender][AllocationType.Seed] += amountToTransfer;
+        }
+        if (userTokens[msg.sender][AllocationType.Private] > 0) {
+            uint256 unlockAmount = _calculateUnlock(
+                msg.sender,
+                AllocationType.Private
+            );
+            rewardsPaid[msg.sender][AllocationType.Private] += unlockAmount;
+            amountToTransfer += unlockAmount;
+        }
+        require(amountToTransfer > 0, "Amount is zero");
+        rewardToken.safeTransfer(msg.sender, amountToTransfer);
+        emit Harvest(msg.sender, amountToTransfer);
+    }
+
+    /// @notice Compute unlocking amount of reward tokens to recipient
+    /// @dev Compute unlocking amount of reward tokens to recipient
+    /// @param _addr the address of investor
+    /// @return the unlocking amount of reward tokens
+    function _calculateUnlock(address _addr, AllocationType _allocationType)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 tokenAmount = userTokens[_addr][_allocationType];
+        uint256 oldRewards = rewardsPaid[_addr][_allocationType];
+        VestingInfo memory _vestingInfo = vestingInfo[_allocationType];
+
+        if (block.timestamp > initialTimestamp + _vestingInfo.cliffDuration) {
+            uint256 countPeriod = (MAX_INITIAL_PERCENTAGE -
+                _vestingInfo.initialPercentage) / 2;
```

Comment:
count period calculates incorrectly. in case when you will have next : 100 - 65 = 35/ 2 = 17. and tokens will unlock early. because we should have 18 periods, not 17